soroban-sdk/src/env.rs

```diff
@@ -469,6 +469,79 @@ use std::{path::Path, rc::Rc};
 #[cfg(any(test, feature = "testutils"))]
 use xdr::{LedgerEntry, LedgerKey, LedgerKeyContractData, SorobanAuthorizationEntry};
 
+#[cfg(any(test, feature = "testutils"))]
+pub struct TestStellarAssetContractUtil {
+    pub address: Address,
+    env: Env,
+    issuer_id: xdr::AccountId,
+}
+
+#[cfg(any(test, feature = "testutils"))]
+impl TestStellarAssetContractUtil {
+    pub fn get_issuer_flags(&self) -> u32 {
+        self.env
+            .host()
+            .with_mut_storage(|storage| {
+                let k = Rc::new(xdr::LedgerKey::Account(xdr::LedgerKeyAccount {
+                    account_id: self.issuer_id.clone(),
+                }));
+
+                let entry = storage.get(
+                    &k,
+                    soroban_env_host::budget::AsBudget::as_budget(self.env.host()),
+                )?;
+
+                match entry.data {
+                    xdr::LedgerEntryData::Account(ref e) => Ok(e.flags.clone()),
+                    _ => panic!("Expected account entry"),
+                }
+            })
+            .unwrap()
+    }
+    /// Sets the issuer flags field.
+    /// Each flag is a bit with values corresponding to [xdr::AccountFlags]
+    ///
+    /// Use this to test interactions between trustlines/balances and the issuer flags.
+    pub fn overwrite_issuer_flags(&self, flags: u32) {
+        if u64::from(flags) > xdr::MASK_ACCOUNT_FLAGS_V17 {
+            panic!(
+                "issuer flags value must be at most {}",
+                xdr::MASK_ACCOUNT_FLAGS_V17
+            );
+        }
+
+        self.env
+            .host()
+            .with_mut_storage(|storage| {
+                let k = Rc::new(xdr::LedgerKey::Account(xdr::LedgerKeyAccount {
+                    account_id: self.issuer_id.clone(),
+                }));
+
+                let mut entry = storage
+                    .get(
+                        &k,
+                        soroban_env_host::budget::AsBudget::as_budget(self.env.host()),
+                    )?
+                    .as_ref()
+                    .clone();
+
+                match entry.data {
+                    xdr::LedgerEntryData::Account(ref mut e) => e.flags = flags,
+                    _ => panic!("Expected account entry"),
```

Comment:
```suggestion
                    _ => panic!("expected account entry"),
```
Same comments as above.

Reply:
done